pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,208 @@
+		}
+
+		return cmd.Process.Pid, nil
+	}
+
+	tests := map[string]utilstest.Test[types.Event, int]{
+		"captures_all_events_without_configured_filters": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				return createTracer(t, &tracer.Config{}, eventCallback)
+			},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectAtLeastOneEvent(func(info *utilstest.Info, ret *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(*ret),
+					Ppid:      uint32(info.Pid),
+					UID:       uint32(info.UID),
+					MountNsID: info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			}),
+		},
+		"filtering_no_event_should_be_captured": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, 0),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectNoEvent[types.Event, int],
+		},
+		"filtering_one_event_should_be_captured": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectOneEvent(func(info *utilstest.Info, ret *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(*ret),
+					Ppid:      uint32(info.Pid),
+					UID:       uint32(info.UID),
+					MountNsID: info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			}),
+		},
+		"event_has_correct_UID": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			// 1435 is just a random unprivileged UID,
+			// there's nothing special about this number.
+			RunnerConfig:  &utilstest.Config{UID: 1435},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectOneEvent(func(info *utilstest.Info, ret *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(*ret),
+					Ppid:      uint32(info.Pid),
+					UID:       uint32(info.UID),
+					MountNsID: info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			}),
+		},
+		"trace_captures_maximum_args": {
```

Comment:
`TestTracer/trace_captures_maximum_args` is also ambiguous. If the behavior is to truncate the args, can we mention it here? Or mention that tracer shows at most 19 arguments?

Reply:
I wonder if renaming it to `args_are_truncated` helps?

Reply:
Literally to `args_are_truncated`, not really, but something like `TestTracer/truncates_captured_args_in_trace_to_maximum_possible_length` or something would sound good to me. I don't think we should be afraid being verbose here. It's fantastic documentation to understand what the code does.